site/components/DateSelector.tsx

```diff
@@ -1,81 +1,124 @@
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-2 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-day-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}${startOrEnd}DateDay`}
-                        type="text"
-                        defaultValue={inputs?.dayInput}
-                    />
-                </div>
-            </div>
-
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-month-input` : `${startOrEnd}-month-input`}
-                    >
-                        Month
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-2 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-month-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateMonth`}
-                        type="text"
-                        defaultValue={inputs?.monthInput}
-                    />
-                </div>
-            </div>
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-year-input` : `${startOrEnd}-year-input`}
-                    >
-                        Year
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-4 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-year-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateYear`}
-                        type="text"
-                        defaultValue={inputs?.yearInput}
-                    />
-                </div>
+const inputBox = (
+    startOrEnd: "start" | "end",
+    hasErrors: boolean,
+    inputRef: React.Ref<any> | undefined,
+    inputProps: InputBaseComponentProps | undefined,
+    InputProps: Partial<FilledInputProps> | Partial<OutlinedInputProps> | undefined,
+    disabled: boolean,
+    isPublishInfo: boolean
+) => (
+    <div className="govuk-date-input flex items-center" id={`${startOrEnd}-date`}>
+        <div className="govuk-date-input__item govuk-!-margin-right-0">
+            <div className="govuk-form-group">
+                <input
```

Comment:
Since we only have a single input, the wrapping fieldset in `DisruptionDateTimeInfo.tsx` should just be a form group with a label inside instead

Reply:
So in `DisruptionDateTimeInfo.tsx`we have the following for the start date and end date:
```
<fieldset
    className="govuk-fieldset"
    role="group"
    aria-describedby={`${isDisruptionValidity ? "validity" : "publish"}-start-date-hint`}
>
    <legend className="govuk-fieldset__legend govuk-!-padding-top-2">
        <h3 className="govuk-heading-s govuk-!-margin-bottom-0">What is the start date?</h3>
    </legend>
    <div id={`${isDisruptionValidity ? "validity" : "publish"}-start-date-hint`} className="govuk-hint">
        Enter in format DD/MM/YYYY
    </div>
    <DateSelector
        errors={[]}
        startOrEnd="start"
        input={!!inputs.validityStartDate ? new Date(inputs.validityStartDate) : null}
        disabled={false}
        isPublishInfo={!isDisruptionValidity}
    />
</fieldset>
```

but now since we only have a single field this can be changed to just be a label rather than a fieldset and legend to something like:

```
<div className="govuk-form-group">
    <label
        className="govuk-label govuk-label--s"
        htmlFor={`${isDisruptionValidity ? "validity" : "publish"}-start-date`}
    >
        What is the start date?
    </label>
    <div id={`${isDisruptionValidity ? "validity" : "publish"}-start-date-hint`} className="govuk-hint">
        Enter in format DD/MM/YYYY
    </div>
    <DateSelector
        errors={[]}
        startOrEnd="start"
        input={!!inputs.validityStartDate ? new Date(inputs.validityStartDate) : null}
        disabled={false}
        isPublishInfo={!isDisruptionValidity}
    />
</div>
```

we just need to make sure the value passed into `htmlFor` on the label matches the id of the input and that we also pass an `aria-describedby` to the input that matches the hint id

Reply:
There is a form-group inside the DateSelector - does that need removing? Or is it normal to have a form group inside a form group?

Reply:
It might actually be nicer if rather than passing through `isPublishInfo`, we could just pass through the id we want to apply to the input, then the DateSelector doesn't need to bother checking if it's publish or not

Reply:
> There is a form-group inside the DateSelector - does that need removing? Or is it normal to have a form group inside a form group?

It's fine to nest them yeah

Reply:
ok cool! will make both changes.